airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -45,6 +50,42 @@
 _G_APP_CRED_ENV_VAR = "GOOGLE_APPLICATION_CREDENTIALS"
 
 
+# Constants used by the mechanism of repeating requests in reaction to exceeding the temporary quota.
+INVALID_KEYS = [
+    'DefaultRequestsPerMinutePerProject',
+    'DefaultRequestsPerMinutePerUser',
+    'RequestsPerMinutePerProject',
+    "Resource has been exhausted (e.g. check quota).",
+]
+INVALID_REASONS = [
+    'userRateLimitExceeded',
+]
+
+
+# pylint: disable=invalid-name
+class retry_if_temporary_quota(tenacity.retry_if_exception):
+    """Retries if there was an exception for exceeding the temporary quote limit."""
+
+    def __init__(self):
+        def is_soft_quota_exception(exception):
```

Comment:
What is the reason to keep this function in constructor?

Reply:
It was written quickly. I still have to improve this code.
My work is based on:
https://github.com/jd/tenacity/blob/master/tenacity/retry.py#L121-L146
This code used the function inside the method